Cache/IO.lean

```diff
@@ -136,6 +148,38 @@ def validateCurl : IO Bool := do
     | _ => throw $ IO.userError "Invalidly formatted version of `curl`"
   | _ => throw $ IO.userError "Invalidly formatted response from `curl --version`"
 
+def validateLeanTar : IO Unit := do
+  if (← LEANTARBIN.pathExists) then return
+  if let some version ← some <$> runCmd "leantar" #["--version"] <|> pure none then
+    let "leantar" :: v :: _ := version.splitOn " "
+      | throw $ IO.userError "Invalidly formatted response from `leantar --version`"
+    let maj :: min :: patch :: _ := v.splitOn "."
+      | throw $ IO.userError "Invalidly formatted version of `leantar`"
+    let version := (maj.toNat!, min.toNat!, patch.toNat!)
+    -- let _ := @lexOrd
+    -- let _ := @leOfOrd
```

Comment:
```suggestion
```

Reply:
these are here on purpose, to be uncommented in the event that the `=` becomes a `>=`. But I've since refactored this part anyway.